lib/rubygems/package/tar_reader/entry.rb

```diff
@@ -150,4 +150,28 @@ def rewind
     @read = 0
   end
 
+private
+
+  ##
+  # Returns the number of bytes in +str+
+  def string_bytesize( str )
+    if str.respond_to? :bytesize then
+      str.bytesize
+    else
+      str.size
+    end
+  end
```

Comment:
Instead of doing this on every call, we should do the test once then define the method dynamically.  For example:

```ruby
if "<3".respond_to? :bytesize
  def string_bytesize str; str.bytesize; end
else
  def string_bytesize str; str.size; end
end
```

This way we only have to do the conditional once.